packages/treasury/src/liquidateMinimum.js

```diff
@@ -45,39 +45,37 @@ async function start(zcf) {
         debtorSeat,
       );
 
-      // if swapOut failed to make the trade, we'll sell it all
-      async function onSwapOutNoTrade() {
-        const strategy = makeDefaultLiquidationStrategy(amm);
-        trace(`onSwapoutFail`);
-
-        const {
-          deposited: sellAllDeposited,
-          userSeatPromise: sellAllSeat,
-        } = await offerTo(
-          zcf,
-          strategy.makeInvitation(runDebt),
-          undefined, // The keywords were mapped already
-          strategy.makeProposal(amountIn, AmountMath.makeEmpty(runBrand)),
-          debtorSeat,
-        );
-        // await sellAllDeposited, but don't need the value
-        await Promise.all([
-          E(sellAllSeat).getOfferResult(),
-          sellAllDeposited,
-        ]).catch(sellAllError => {
-          throw Error(`Unable to liquidate ${sellAllError}`);
-        });
-      }
-
       // await deposited, but we don't need the value. We'll need it to have
       // resolved in both branches, so can't put it in Promise.all.
       await deposited;
       await E(liqSeat).getOfferResult();
```

Comment:
I don't understand why `Promise.all` doesn't work; is there an ordering constraint between the two?

```suggestion
      await Promise.all([deposited, E(liqSeat).getOfferResult()]);
```

Reply:
There is an ordering constraint. deposited is a promise that resolves when `offerTo`'s transfer is complete, and thus it's safe to call `getOfferResult`.

https://agoric.com/documentation/zoe/api/zoe-helpers.html#offerto-zcf-invitation-keywordmapping-proposal-fromseat-toseat

Reply:
OK; I didn't get that from "We'll need it to have resolved in both branches"; maybe rephrase the code comment to state the ordering constraint as you did here?